### ISSUE TITLE
choices not checked on global option when using commands

### ISSUE THREAD

**Issue by tacomanator** [yargs/yargs#577]
I may be misunderstanding the config, but I am expecting an a global option to be checked for a valid choice even when using commands:

```javascript
// script.js
argv = require('yargs')
  .option('size', {
    alias: 's',
    describe: 'choose a size',
    choices: ['xs', 's', 'm', 'l', 'xl']
    global: true
  })
  .command('foo')
  .command('bar')
  .help('help')
  .argv;

console.dir(argv);
```

valid command but invalid size, no help shown:

```bash
$ node script.js -s nope foo
{ _: [ 'foo' ],
  help: false,
  s: 'nope',
  size: 'nope',
  '$0': 'script.js' }
```

now with invalid command, still invalid size, this time help shown for invalid size:

```bash
$ node script.js -s nope baz
Commands:
  foo
  bar

Options:
  --size, -s  choose a size                 [choices: "xs", "s", "m", "l", "xl"]
  --help      Show help                                                [boolean]

Invalid values:
  Argument: size, Given: "nope", Choices: "xs", "s", "m", "l", "xl"
```

Is this expected? Using 4.8.1.

**Comment by nexdrew** [comment]
@tacomanator Unfortunately, yargs will only reset, reparse, and perform validation when given a minimal builder object or function, which I [recently discovered](https://github.com/yargs/yargs/pull/567#issuecomment-234270884) 😞. According to #492, we may want to consider this a bug (or at least that it's less than ideal).

The current solution/workaround for this is to apply `{}` as the third argument to `.command()`, like so:

```js
var argv = require('yargs')
  .option('size', {
    alias: 's',
    describe: 'choose a size',
    choices: ['xs', 's', 'm', 'l', 'xl'],
    global: true
  })
  .command('foo', 'foo description', {}) // <-- "runnable" command
  .command('bar', false, {}) // <-- "hidden" command
  .help('help')
  .argv

console.dir(argv)
```

**Comment by tacomanator** [comment]
Thanks for the workaround @nexdrew.

On a side note, in case anyone refers to this issue I would like to chime in that I believe global arguments should be global. In other words, `node script.js -s xl foo` should be the same as `node script.js foo -s xl` for the global argument `s`. In this way arguments can be passed to the script even when forwarded from a wrapper such as `npm run script-wrapper -- -s xl`.

**Comment by nexdrew** [comment]
This should be fixed by #583. 🎉

**Comment by bcoe** [comment]
@tacomanator `yargs@5.x` is out the door, so you should be able to try out the fix in #583 

Simply run:

`npm cache clear; npm i yargs@next`

Feel free to reopen this ticket if you continue running into problems.

With regards to "global arguments being global", mind opening a new issue on this topic so we don't lose track?
